### ISSUE TITLE
SvelteKit v2: $page not reevaluated after invalidateAll

### ISSUE THREAD

**Issue by peterpeterparker** [sveltejs/kit#11417]
### Describe the bug

 In SvelteKit v2, the `$page` derived store is not reevaluated after triggering `invalidateAll`. For example, modifying the URL using `replaceState` and then calling `invalidateAll` to propagate the new URL parameters does not work as expected.

This behavior is different from SvelteKit v1, and I assume it's not an intentional breaking change.

---

Snippet:

`+page.svelte`

```
<script lang="ts">
	import { page } from '$app/stores';
	import { invalidateAll, replaceState } from '$app/navigation';

	const updateUrl = async () => {
		const url = new URL(window.location.href);
		url.searchParams.set('network', `${Math.random()}`);
		replaceState(url, {});

		await invalidateAll();
	}
</script>

<output>{$page.data.network}</output>

<button on:click={updateUrl}>Update URL</button>
```

`+page.ts`

```
import type { LoadEvent } from '@sveltejs/kit';

const loadRouteParams = ($event: LoadEvent) => {
	const {
		url: { searchParams }
	} = $event;

	return {
		network: searchParams?.get('network'),
	};
};

export const load = ($event: LoadEvent) => loadRouteParams($event);
```

<img width="996" alt="Capture d’écran 2023-12-20 à 16 26 04" src="https://github.com/sveltejs/kit/assets/16886711/b420a96a-8da1-4969-a527-5c468eb342c8">

### Reproduction

git clone https://github.com/peterpeterparker/invalidate
cd invalidate
npm ci
npm run dev

### System Info

```Shell
System:
    OS: macOS 14.1.1
    CPU: (8) arm64 Apple M2
    Memory: 125.02 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.16.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 9.6.1 - /opt/homebrew/bin/npm
  Browsers:
    Brave Browser: 120.1.61.104
    Chrome: 120.0.6099.109
    Firefox: 121.0
    Safari: 17.1
  npmPackages:
    @sveltejs/adapter-auto: ^3.0.1 => 3.0.1 
    @sveltejs/kit: ^2.0.4 => 2.0.4 
    @sveltejs/vite-plugin-svelte: ^3.0.1 => 3.0.1 
    svelte: ^4.2.8 => 4.2.8 
    vite: ^5.0.3 => 5.0.10
```

### Severity

blocking an upgrade

**Comment by mostrecent** [comment]
I've faced something similar, not sure if it is related: https://github.com/sveltejs/kit/issues/11415

**Comment by PatrickG** [comment]
`pushState` and `replaceState` are for shallow routes only.
After using them, Sveltekits point of view is still the previous page. Only `$page.state` is updated.

You can use `goto` with `{ replaceState: true }` like in Sveltekit v1.

**Comment by iolyd** [comment]
@peterpeterparker, a bit out of subject, but concerning your point of losing the scroll position: have you tried
```ts
goto(..., {
  replaceState: true,
  noScroll: true
})
```

**Comment by peterpeterparker** [comment]
@iolyd yeah I did but it did not work as excepted, I might have screw something. Let me try again...

**Comment by peterpeterparker** [comment]
of course it works now! Never try stuff late at night, I guess issue was between screen and keyboard.

I'll do a `await goto(url, { replaceState: true, noScroll: true })` and I'll be fine, so I can close this issue.

Thanks @iolyd and my bad then @PatrickG that was a false assumption on the scroll.

**Comment by PatrickG** [comment]
> of course it works now! Never try stuff late at night, I guess issue was between screen and keyboard.
> 
> I'll do a `await goto(url, { replaceState: true, noScroll: true })` and I'll be fine, so I can close this issue.
> 
> Thanks @iolyd and my bad then @PatrickG that was a false assumption on the scroll.

Glad you could solve your issue.
I would suggest to add `keepFocus: true` as well.
